src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_LockStatement.cs

```diff
@@ -37,6 +40,41 @@ public override BoundNode VisitLockStatement(BoundLockStatement node)
                     argumentType); //need to have a non-null type here for TempHelpers.StoreToTemp.
             }
 
+            if (argumentType.IsWellKnownTypeLock() &&
+                TryGetWellKnownTypeMember(lockSyntax, WellKnownMember.System_Threading_Lock__EnterLockScope, out MethodSymbol enterLockScope) &&
+                TryGetWellKnownTypeMember(lockSyntax, WellKnownMember.System_Threading_Lock__Scope__Dispose, out MethodSymbol lockScopeDispose))
```

Comment:
Can these calls to `TryGetWellKnownTypeMember()` fail here? If so, should we generate `BoundBadExpression` rather than falling through to the `Monitor.Enter` case?